lib/nostrum/api.ex

```diff
@@ -276,49 +277,95 @@ defmodule Nostrum.Api do
   end
 
   @doc ~S"""
-  Create a rection for a message.
+  Creates a reaction for a message.
 
-  Creates a reaction using an `emoji` for the message specified by `message_id` and
-  `channel_id`. `emoji` can be a `Nostrum.Struct.Emoji.custom_emoji.t`, a
-  base 16 unicode emoji string, or a uri encoded string.
+  If the request was successful, this function returns `{:ok}`. Otherwise, 
+  this function returns `{:error, reason}`.
+
+  ## Examples
 
-  **Example**
   ```Elixir
+  # Using a Nostrum.Struct.Emoji.
+  emoji = %Nostrum.Struct.Emoji{id: 43819043108, name: "foxbot"}
+  Nostrum.Api.create_reaction(123123123123, 321321321321, emoji)
+
+  # Using a base 16 emoji string.
   Nostrum.Api.create_reaction(123123123123, 321321321321, "\xF0\x9F\x98\x81")
+
+  # Using a URI encoded emoji string.
   Nostrum.Api.create_reaction(123123123123, 321321321321, URI.encode("\u2b50"))
   ```
 
-  Returns `{:ok}` if successful, `{:error, reason}` otherwise.
+  For other emoji string examples, see `t:Nostrum.Struct.Emoji.emoji_api_name/0`.
   """
-  @spec create_reaction(integer, integer, String.t | Nostrum.Struct.Emoji.custom_emoji) :: error | {:ok}
-  def create_reaction(channel_id, message_id, emoji) do
-    request(:put, Constants.channel_reaction_me(channel_id, message_id, emoji))
+  @spec create_reaction(Channel.id, Message.id, Emoji.t | Emoji.emoji_api_name) :: error | {:ok}
```

Comment:
Maybe we should include an `emoji` type at the top of this module that is comprised of this definition here? Namely the `Emoji.t | Emoji.emoji_api_name` portion.

Reply:
Agreed. I'll be adding that in this upcoming commit.